testTmpExists fails if TMPDIR != /tmp on Linux

**Describe the bug**
When `TMPDIR` is not `/tmp` (or a subdirectory of `/tmp`), the following test failure occurs:

```pytb
======================================================================
FAIL: testTmpExists (pyfakefs.tests.fake_filesystem_unittest_test.TestTempDirCreation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/pyfakefs/pyfakefs/tests/fake_filesystem_unittest_test.py", line 666, in testTmpExists
    self.assertTrue(os.path.exists("/tmp"))
AssertionError: False is not true
```

I think this is because https://github.com/pytest-dev/pyfakefs/blob/main/pyfakefs/fake_filesystem_unittest.py#L894-L903 only creates `/tmp` symlink on Darwin (and only `TMPDIR` on Linux) but https://github.com/pytest-dev/pyfakefs/blob/main/pyfakefs/tests/fake_filesystem_unittest_test.py#L663-L666 assumes `/tmp` on all non-win32 platforms.

**How To Reproduce**
```
export TMPDIR=/var/tmp
tox
```

**Your environment**
Please run the following in the environment where the problem happened and
paste the output.
```bash
python -c "import platform; print(platform.platform())"
python -c "import sys; print('Python', sys.version)"
python -c "from pyfakefs.fake_filesystem import __version__; print('pyfakefs', __version__)"
python -c "import pytest; print('pytest', pytest.__version__)"
```

```
Linux-6.2.10-gentoo-dist-x86_64-AMD_Ryzen_5_3600_6-Core_Processor-with-glibc2.37
Python 3.11.3 (main, Apr  6 2023, 04:25:37) [GCC 12.2.1 20230304]
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name '__version__' from 'pyfakefs.fake_filesystem' (/tmp/pyfakefs/pyfakefs/fake_filesystem.py)
pytest 7.3.0
```

(git 1906fde13c29de1ca4c232177490c39aae197aa8)

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -13,6 +13,8 @@ The released versions correspond to PyPI releases.
 * Some public constants in `fake_filesystem` that had been moved to `helpers` are
   made accessible from there again (see [#809](../../issues/809)).
 * Add missing fake implementations for `os.getuid` and `os.getgid` (Posix only)
+* Make sure a `/tmp` path exists under linux (`TMPDIR` may point elsewhere)
+  (see [#810](../../issues/810))
 
 
 ## [Version 5.2.1](https://pypi.python.org/pypi/pyfakefs/5.2.1) (2023-04-11)

--- a/docs/troubleshooting.rst
+++ b/docs/troubleshooting.rst
@@ -170,8 +170,12 @@ directory named ``/tmp`` when running on Linux or Unix systems,
       # the temp directory is always present at test start
       assert len(os.listdir("/")) == 1
 
-Under macOS, a symlink to the actual temp directory is created additionally as `/tmp`
-in the fake filesystem.
+Under macOS and linux, if the actual temp path is not `/tmp` (which is always the case
+under macOS), a symlink to the actual temp directory is additionally created as `/tmp`
+in the fake filesystem. Note that the file size of this link is ignored while
+calculating the fake filesystem size, so that the used size with an otherwise empty
+fake filesystem can always be assumed to be 0.
+
 
 User rights
 -----------

--- a/pyfakefs/fake_filesystem_unittest.py
+++ b/pyfakefs/fake_filesystem_unittest.py
@@ -895,8 +895,8 @@ class Patcher:
         # so we create it here for convenience
         assert self.fs is not None
         self.fs.create_dir(temp_dir)
-        if sys.platform == "darwin" and not self.fs.exists("/tmp"):
-            # under macOS, we also create a link in /tmp for convenience
+        if sys.platform != "win32" and not self.fs.exists("/tmp"):
+            # under Posix, we also create a link in /tmp if the path does not exist
             self.fs.create_symlink("/tmp", temp_dir)
             # reset the used size to 0 to avoid having the link size counted
             # which would make disk size tests more complicated

--- a/pyfakefs/tests/fake_filesystem_unittest_test.py
+++ b/pyfakefs/tests/fake_filesystem_unittest_test.py
@@ -662,7 +662,7 @@ class TestTempDirCreation(fake_filesystem_unittest.TestCase):
 
     @unittest.skipIf(sys.platform == "win32", "POSIX only test")
     def testTmpExists(self):
-        # directory under Linux, link under macOS
+        # directory or link under Linux, link under macOS
         self.assertTrue(os.path.exists("/tmp"))
 
 

